src/Microsoft.AspNet.Mvc.ModelBinding/Binders/MutableObjectModelBinder.cs

```diff
@@ -482,11 +454,20 @@ internal void ProcessDto(ModelBindingContext bindingContext, ComplexModelDto dto
             else
             {
                 // trying to set a non-nullable value type to null, need to make sure there's a message
-                var modelStateKey = dtoResult.ValidationNode.ModelStateKey;
+                var modelStateKey = dtoResult.ModelStateKey;
                 var validationState = bindingContext.ModelState.GetFieldValidationState(modelStateKey);
                 if (validationState == ModelValidationState.Unvalidated)
                 {
-                    dtoResult.ValidationNode.Validated += CreateNullCheckFailedHandler(propertyMetadata, value);
+                    //dtoResult.ValidationNode.Validated += CreateNullCheckFailedHandler(propertyMetadata, value);
+                    // TODO: https://github.com/aspnet/Mvc/issues/450 Revive ModelBinderConfig
+                    // var errorMessage =  ModelBinderConfig.ValueRequiredErrorMessageProvider(e.ValidationContext,
+                    //                                                                            modelMetadata,
```

Comment:
this was left as is intentionally, I did not want to remove the TODO, but now looking back i feel this can be erased.